src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -110,7 +110,11 @@ def forward(self, x, position_ids, seq_len=None):
 
         inv_freq_expanded = self.inv_freq[None, :, None].float().expand(position_ids.shape[0], -1, 1)
         position_ids_expanded = position_ids[:, None, :].float()
-        freqs = (inv_freq_expanded @ position_ids_expanded).transpose(1, 2)
+        
+        # Force float32 since bfloat16 loses precision on long contexts
+        with torch.autocast(device_type=position_ids_expanded.device.type, enabled=False):
+            freqs = (inv_freq_expanded.float() @ position_ids_expanded.float()).transpose(1, 2)
```

Comment:
```suggestion
        with torch.no_grad():
            freqs = (inv_freq_expanded.float() @ position_ids_expanded.float()).transpose(1, 2)
```
we don't need the gradients actually no? That would be the best fix!

Reply:
Oh ye we don't need gradients - in fact the entire `forward` can be `torch.inference_mode()` :)

Reply:
let's add that as well!

Reply:
:) Wrapped the entire forward :)